### ISSUE TITLE
Improve performance when dealing with transactions

### ISSUE THREAD

**Issue by 4miners** [LiskHQ/lisk#1870]
### Expected behavior
`modules.transactions.fillPool` should be called as part of `forge`, probably best place is here: https://github.com/LiskHQ/lisk/blob/9664c2b5556fb663c98f07a1f8c5c343c44e57e6/modules/delegates.js#L250
So `modules.transactions.fillPool` will only be executed just before forging actual block. It will should improve performance noticeably.
### Actual behavior
`modules.transactions.fillPool` is called after `forge` (wrong logic) and every time when `nextForge` is triggered:
https://github.com/LiskHQ/lisk/blob/9664c2b5556fb663c98f07a1f8c5c343c44e57e6/modules/delegates.js#L921
### Steps to reproduce
N/A
### Which version(s) does this affect? (Environment, OS, etc...)
1.0.0 / 0.9.x

**Comment by simonmorgenthaler** [comment]
Under heavy load when every block should be full, sometimes empty blocks are forged. Might be this issue?
Happened also with beta9 on betanet.

![clipboard - june 18 2018 2_49 pm](https://user-images.githubusercontent.com/10214886/41539391-f8f26f3a-730d-11e8-83c1-8e035017c56e.png)
![clipboard - june 18 2018 2_47 pm](https://user-images.githubusercontent.com/10214886/41539392-f92a2a92-730d-11e8-996e-58632d5ff95a.png)

**Comment by 4miners** [comment]
@simonmorgenthaler Yep, the reason of that behavior is calling `fillPool` after `forge` instead of before. We already tested it and after reverse of the order - there are no more empty blocks. However that change is not planned for `1.0.0` release.

**Comment by MaciejBaj** [comment]
@4miners please open the following issues.

**Comment by 4miners** [comment]
For `1.1.0` we will only do https://github.com/LiskHQ/lisk/issues/2170, other things described in this issue and comments are to review when redesigning/refactoring/rewriting transaction pool.

**Comment by MaciejBaj** [comment]
@4miners:
> there is no need to perform applyUnconfirmed on pending transactions when node is not forging

`applyUnconfirmed` step on non-forging nodes cannot be removed from `fillPool`. Peer would broadcast all of the pending transactions possibly polluting the network with invalid ones. More details is described here - https://github.com/LiskHQ/lisk/issues/2209#issuecomment-405024388.

The problem of empty blocks while having a full transaction pool -  https://github.com/LiskHQ/lisk/issues/1870#issuecomment-398057675 is fixed by #2171.

Issue can be closed.
